### ISSUE TITLE
`ytt fmt` does not handle percentage signs  

### ISSUE THREAD

**Issue by epk** [k14s/ytt#370]
**What steps did you take:**

```bash
$ cat test.yaml               
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    name: jobs-low-2
    type: jobs
  name: jobs-low-2
spec:
  progressDeadlineSeconds: 900
  revisionHistoryLimit: 10
  strategy:
    type: RollingUpdate
    rollingUpdate:
      maxSurge: 0
      maxUnavailable: 25%

$ cat test.yaml | ytt fmt -f -
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    name: jobs-low-2
    type: jobs
  name: jobs-low-2
spec:
  progressDeadlineSeconds: 900
  revisionHistoryLimit: 10
  strategy:
    type: RollingUpdate
    rollingUpdate:
      maxSurge: 0
      maxUnavailable: 25%!(NOVERB)
```

**What happened:**
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    name: jobs-low-2
    type: jobs
  name: jobs-low-2
spec:
  progressDeadlineSeconds: 900
  revisionHistoryLimit: 10
  strategy:
    type: RollingUpdate
    rollingUpdate:
      maxSurge: 0
      maxUnavailable: 25%!(NOVERB) <- ???
```

**What did you expect:**

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    name: jobs-low-2
    type: jobs
  name: jobs-low-2
spec:
  progressDeadlineSeconds: 900
  revisionHistoryLimit: 10
  strategy:
    type: RollingUpdate
    rollingUpdate:
      maxSurge: 0
      maxUnavailable: 25%
```

**Anything else you would like to add:**

I think we are not properly escaping `%` in the parser

**Environment:**

- ytt version (use `ytt --version`): ytt version 0.32.0

- OS (e.g. from `/etc/os-release`): macOS 11.2.3 (20D91)

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

**Comment by pivotaljohn** [comment]
Thanks for reporting this bug, @epk.

To be clear, the `%!(NOVERB)` is output from `fmt.Fprintf()`. Which indicates that we're passing the raw string to that function without escaping the `%`, first, as @epk said.

Note that we're doing the proper handling in the `template` command:

```console
$ ytt -f test.yml
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    name: jobs-low-2
    type: jobs
  name: jobs-low-2
spec:
  progressDeadlineSeconds: 900
  revisionHistoryLimit: 10
  strategy:
    type: RollingUpdate
    rollingUpdate:
      maxSurge: 0
      maxUnavailable: 25%
```

**Comment by aaronshurley** [comment]
This feels like a reasonable `good first issue` to me. @pivotaljohn please correct this if I'm wrong.

We're going to move this to our Unprioritized Backlog at the moment but are open to PRs!

**Comment by cari-lynn** [comment]
Available in [v0.38.0](https://github.com/vmware-tanzu/carvel-ytt/releases/tag/v0.38.0)
